### PR TITLE
GH-2313: Add checkout paths to routes default config

### PR DESCRIPTION
**Pull request by Matejk00**
Closes: #2313

### CODE REVIEW

File: projects/core/src/routing/configurable-routes/config/default-storefront-routes-config.ts [hunk 1]
```diff
@@ -8,7 +8,10 @@ export const defaultStorefrontRoutesConfig: RoutesConfig = {
   register: { paths: ['register'] },
   resetPassword: { paths: ['login/pw/change'] },
   forgotPassword: { paths: ['forgot-password'] },
-  checkout: { paths: ['checkout'] },
+  shippingAddress: { paths: ['checkout/shipping-address'] },
+  deliveryMode: { paths: ['checkout/delivery-mode'] },
+  paymentDetails: { paths: ['checkout/payment-details'] },
+  reviewOrder: { paths: ['checkout/review-order'] },
   orderConfirmation: { paths: ['order-confirmation'] },
```
Comment (Matejk00) [inline projects/core/src/routing/configurable-routes/config/default-storefront-routes-config.ts]: We need to figure out how we can combine our CheckoutConfig routes from defaultCheckoutConfig with this file to setup routing dynamically. This config file is on a core side.

Comment (Matejk00) [reply projects/core/src/routing/configurable-routes/config/default-storefront-routes-config.ts]: For now i think we can keep it mocked.

Comment (Matejk00) [reply projects/core/src/routing/configurable-routes/config/default-storefront-routes-config.ts]: This modification refers to following breaking change: https://github.com/SAP/cloud-commerce-spartacus-storefront/pull/1994

File: projects/storefrontlib/src/lib/ui/pages/default-routing-config.ts [hunk 2]
```diff
@@ -8,7 +8,10 @@ export const defaultStorefrontRoutesConfig: RoutesConfig = {
   register: { paths: ['register'] },
   resetPassword: { paths: ['login/pw/change'] },
   forgotPassword: { paths: ['forgot-password'] },
-  checkout: { paths: ['checkout'] },
```
Comment (marlass) [inline projects/storefrontlib/src/lib/ui/pages/default-routing-config.ts]: Should we remove this route? Other components might use it? If those components should link to checkout this one should be used.

Comment (marlass) [reply projects/storefrontlib/src/lib/ui/pages/default-routing-config.ts]: At least 2 components are using this checkout route, so let's keep it.

Comment (Matejk00) [reply projects/storefrontlib/src/lib/ui/pages/default-routing-config.ts]: I reverted that change.

**Pull request merged**